app/src/main/java/com/asfoundation/wallet/service/KeyStoreFileManager.java

```diff
@@ -0,0 +1,111 @@
+package com.asfoundation.wallet.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import org.web3j.crypto.WalletFile;
+
+public class KeyStoreFileManager {
+  private final String keystoreFolderPath;
+  private final ObjectMapper mapper;
+
+  public KeyStoreFileManager(String keystoreFolderPath, ObjectMapper mapper) {
+    this.mapper = mapper;
+    new File(keystoreFolderPath).mkdirs();
+    if (keystoreFolderPath.charAt(keystoreFolderPath.length() - 1) == '/') {
+      this.keystoreFolderPath = keystoreFolderPath;
+    } else {
+      this.keystoreFolderPath = keystoreFolderPath + "/";
+    }
+  }
+
+  public String getKeystore(String accountAddress) {
+    return Objects.requireNonNull(
+        getFilePath(removeHexIndicator(accountAddress), keystoreFolderPath),
+        "Wallet with address: " + accountAddress + " not found");
+  }
+
+  private String removeHexIndicator(String accountAddress) {
+    return accountAddress.replace("0x", "");
+  }
+
+  private String getFilePath(String accountAddress, String path) {
+    File file = new File(path);
+    if (!file.isDirectory()) {
+      if (file.getName()
+          .toLowerCase()
+          .contains(accountAddress.toLowerCase())) {
+        return file.getAbsolutePath();
+      }
+    }
+    if (file.isDirectory()) {
```

Comment:
Should we start with the 'isDirectory' if statement and the use 'else' to check if the file name matches the account address?

Reply:
corrected